### DISCUSSION TITLE
grand-child component not getting re-rendered on error

### DISCUSSION THREAD

**Discussion by saverineni** [react-hook-form/react-hook-form#4100] [Q&A]
Hi all,

I am working on react-hook validations. 

Parent component (1)  encapsulates a form, form contains a component (2) and that component contains couple of text boxes ( components 3 and 4).

form -> component -> 2 text boxes

When there is a validation error in those text boxes I am unable to show field level validations. I assume components 3 and 4 are not getting re-rendered on validation error. 

But i can show form level validations from the top-level component( which has form).

This is an example image i uploaded, where the field level error messages i couldn't able to achieve with my design.

<img width="643" alt="Screenshot 2021-02-03 at 00 09 44" src="https://user-images.githubusercontent.com/7191444/106679571-e1b22800-65b4-11eb-9faa-42c6778f4fca.png">

Any help would be greatly appreciated.

Babu

**Comment by bluebill1049** [comment]
codesandbox?
